dot/core/service_test.go

```diff
@@ -577,12 +651,708 @@ func Test_Service_handleBlocksAsync(t *testing.T) {
 			blockAddChan <- &block
 			close(blockAddChan)
 		}()
-		s := &Service{
+		service := &Service{
 			blockState:       mockBlockState,
 			transactionState: mockTxnStateErr,
 			blockAddCh:       blockAddChan,
 			ctx:              context.Background(),
 		}
-		s.handleBlocksAsync()
+		service.handleBlocksAsync()
+	})
+}
+
+func TestService_handleChainReorg(t *testing.T) {
+	t.Parallel()
+	execTest := func(t *testing.T, s *Service, prevHash common.Hash, currHash common.Hash, expErr error) {
+		err := s.handleChainReorg(prevHash, currHash)
+		assert.ErrorIs(t, err, expErr)
+		if expErr != nil {
+			assert.EqualError(t, err, expErr.Error())
+		}
+	}
+
+	testPrevHash := common.MustHexToHash("0x01")
+	testCurrentHash := common.MustHexToHash("0x02")
+	testAncestorHash := common.MustHexToHash("0x03")
+	testSubChain := []common.Hash{testPrevHash, testCurrentHash, testAncestorHash}
+	ext, externExt, body := generateExtrinsic(t)
+	testValidity := &transaction.Validity{Propagate: true}
+	vtx := transaction.NewValidTransaction(ext, testValidity)
```

Comment:
we do we need all of this if we're just mocking it out anyways? Looks like this is the only reference to `generateExtrinsic`

Reply:
Thanks for the explanation!  Maybe add a comment saying the transaction is checked.